tomcat/src/main/java/org/apache/catalina/http/startline/HttpRequestLine.java

```diff
@@ -0,0 +1,61 @@
+package org.apache.catalina.http.startline;
+
+import java.nio.file.Path;
+import org.apache.catalina.exception.CatalinaException;
+
+public class HttpRequestLine {
+
+    private final HttpMethod httpMethod;
+    private final RequestUri requestUri;
+    private final HttpVersion httpVersion;
+
+    public HttpRequestLine(HttpMethod httpMethod, RequestUri requestUri, HttpVersion httpVersion) {
+        this.httpMethod = httpMethod;
+        this.requestUri = requestUri;
+        this.httpVersion = httpVersion;
+    }
+
+    public static HttpRequestLine parse(String startLine) {
+        String[] separatedStartLine = validateAndSplit(startLine);
+        return new HttpRequestLine(
```

Comment:
HttpRequestLine를 사용하는 입장에서는 아래와 같이 전달하는게 더 편하지 않을까요

```suggestion
        return new HttpRequestLine(separatedStartLine[1], separatedStartLine[2]);
```

Reply:
사용하는 입장에서 String 인자를 3개 받는 생성자가 존재한다면, 실수가 발생할 확률이 높을 것 같아 이와 같이 작성했습니다. 입력을 하는 입장에서는 RequestURI를 제외하곤 Enum을 넣어주면 되기 때문에 이 방법이 더 괜찮아 보였습니다.